tests/test_cvi.py

```diff
@@ -11,41 +11,206 @@
+
+    sample = local_data["samples"][index, :]
+    label = local_data["labels"][index]
+
+    return sample, label
+
 # --------------------------------------------------------------------------- #
 # TESTS
 # --------------------------------------------------------------------------- #
 
 class TestCVI:
 
-    def test_opts(self):
+    def test_load_data(self, data) -> None:
+        """Test loading the partitioning data.
+        """
+
+        lg.info("--- TESTING DATA LOADING ---")
+        lg.info(f"Data location: {id(data)}")
+
+        for value in data["datasets"].values():
+            log_data(value)
+
+        return
+
+    def test_loading_again(self, data:dict) -> None:
+        """
+        Tests loading the data again to verify the identity of the data dictionary.
+        """
+
+        lg.info("--- TESTING LOADING AGAIN TO VERIFY DATA SINGLETON ---")
+        log_data(data["datasets"]["correct"])
+        lg.info(f"Data location: {id(data)}")
+
+        return
+
+    def test_icvis(self, data:dict) -> None:
+        """
+        Test the functionality all of the icvis.
+        """
+
+        lg.info("--- TESTING ALL ICVIS ---")
+
+        tolerance = 1e-1
+
+        for key, local_data in data["datasets"].items():
+            lg.info(f"Testing data: {key}")
+            # Incremental
+            i_cvis = get_cvis()
+            for local_cvi in i_cvis:
+                lg.info(local_cvi)
+                for ix in range(data["counts"][key]):
+                    # Grab a sample and label
+                    sample, label = get_sample(local_data, ix)
+                    local_cvi.param_inc(sample, label)
+                    local_cvi.evaluate()
+            # Batch
+            b_cvis = get_cvis()
+            for local_cvi in b_cvis:
+                local_cvi.param_batch(local_data["samples"], local_data["labels"])
+                local_cvi.evaluate()
+
+            # Test equivalence between batch and incremental results
+            for i in range(len(i_cvis)):
+                assert (i_cvis[i].criterion_value - b_cvis[i].criterion_value) < tolerance
+                lg.info(f"I: {i_cvis[i].criterion_value}, B: {b_cvis[i].criterion_value}")
+
+        return
+
+    def test_opts(self) -> None:
+        """
+        Test the CVI options object.
+        """
+
         my_opts = cvi.CVIOpts()
         lg.info(my_opts)
-        return
 
-    def test_cvi(self):
-        my_cvi =cvi.modules.CH()
-        lg.info(my_cvi)
         return
 
-    def test_all_cvis(sefl):
+    def test_setup_icvi(self) -> None:
+        """
+        Test running the setup method on ICVIs.
+        """
+
+        # Create the cvi/icvi module
         my_cvi = cvi.modules.CH()
+
+        lg.info("Before setup")
+        lg.info(my_cvi)
+
         sample = np.ones(3)
         my_cvi.setup(sample)
-        lg.info("asdf")
+
+        lg.info("After setup")
         lg.info(my_cvi)
-        return
\ No newline at end of file
+
+        return
```

Comment:
In the future lets add a test to not only verify the batch and incremental versions have the same results but also that batch is slower than the incremental version

Reply:
That would be a good test. I'll make that an issue and new feature.

Reply:
Issue #5